File: pipeline/calculate_alignment.py
"""
calculate_alignment.py

Does nothing and then calls  renderer Lambda
"""
from pathlib import Path
from urllib.parse import unquote

import os
import json

import boto3

def main(event, context):
    '''Does nothing'''

    # extract key and source bucket from incoming event
    key = unquote(event['Records'][0]['s3']['object']['key'])
    choir_id, song_id,part_id  = Path(key).stem.split('.')[0].split('+')
    bucket = event['Records'][0]['s3']['bucket']['name']
    print('Running on %s/%s' % (bucket, key))

    lambda_client = boto3.client('lambda')

    ret = {"key": key,
           "bucket": bucket,
           "choir_id": choir_id,
           "song_id": song_id,
           "part_id": part_id}

    lambda_client.invoke(
	FunctionName= os.environ['RENDERER_LAMBDA'],
	Payload = json.dumps(ret),
	InvocationType = 'Event'
    )
    return ret

File: pipeline/calculate_alignment.js
// environment variables
const RENDERER_LAMBDA = process.env.RENDERER_LAMBDA

// node modules
const path = require('path')
const aws = require('aws-sdk')

// aws objects
const Lambda = new aws.Lambda()

const main = async (event, context) => {
  console.log('calculate_alignment')

  // event is an AWS event
  const key = unescape(event.Records[0].s3.object.key)
  const bucket = event.Records[0].s3.bucket.name
  console.log(`Running on ${bucket}/${key}`)

  // invoke next Lambda
  const bits = path.parse(key).name.split('+')
  const payload = {
    key: key,
    bucket: bucket,
    choir_id: bits[0],
    song_id: bits[1],
    part_id: bits[2]
  }
  const params = {
    FunctionName: RENDERER_LAMBDA,
    InvocationType: 'Event',
    Payload: JSON.stringify(payload)
  }
  await Lambda.invoke(params).promise()
}

exports.main = main
